atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -389,26 +406,45 @@ def do_worker_build(self, release, cluster_info, koji_upload_dir, task_id):
                 except OsbsException:
                     pass
 
+    def select_and_start_cluster(self, platform):
+        ''' Choose a cluster and start a build on it '''
+
+        config = get_config(self.workflow)
+        clusters = config.get_enabled_clusters_for_platform(platform)
+
+        if not clusters:
+            raise UnknownPlatformException('No clusters found for platform {}!'
+                                           .format(platform))
+
+        cluster_fails = Counter()
+        retry_at = {}
+        possible_clusters = None
+
+        while True:
+            if not clusters:
+                raise RuntimeError('Could not find appropriate cluster for worker build.')
+            possible_clusters = self.get_clusters(clusters, cluster_fails, platform, retry_at)
+            for cluster in possible_clusters or []:
+                try:
+                    self.do_worker_build(self.release, cluster, self.koji_upload_dir,
+                                         self.fs_task_id)
+                    return
+                except RuntimeError:
```

Comment:
Is this the right exception to catch? It looks to me like do_worker_build doesn't raise an exception if creating the build failed -- it probably should, so we get into this retry loop. (Once created, it shouldn't raise an exception if monitoring fails.)

But we don't necessarily know what exception it will raise so maybe we need to catch `Exception` here? But also have a clause for `BuildCanceledException` to make sure that gets propagated immediately.